pkg/policy/caps.go

```diff
@@ -0,0 +1,36 @@
+package policy
+
+import (
+	"github.com/aquasecurity/tracee/pkg/capabilities"
+	"github.com/aquasecurity/tracee/pkg/errfmt"
+	"github.com/aquasecurity/tracee/pkg/events"
+)
+
+// UpdateCapabilitiesRings updates the capabilities rings based on the policies events states.
+func (ps *Policies) UpdateCapabilitiesRings() error {
+	// NOTE: This only adds capabilities from events that are defined in the policies.
+	// TODO: On the Snapshot pruning, we should check which capabilities are not used
+	// for any snapshot and remove them from the capabilities rings.
+
+	caps := capabilities.GetInstance()
+
+	ps.rwmu.RLock()
+	defer ps.rwmu.RUnlock()
+
+	for id := range ps.evtsStates.states {
```

Comment:
It would cause a deadlock as `Policies.EventsStates()` also requires a lock. That usage makes part of the pattern I'm bringing: whether in thread-safe scope, use the inner private objects not the public (which also makes use of the same control).

Reply:
I see.
You will still need to use the `GetAll()` method though